crates/cast/bin/cmd/storage.rs

```diff
@@ -226,18 +221,17 @@ async fn fetch_and_print_storage(
     }
 }
 
-async fn fetch_storage_slots(
-    provider: RetryProvider,
-    address: NameOrAddress,
+async fn fetch_storage_slots<P: Provider<Ethereum, T>, T: Transport + Clone>(
+    provider: P,
```

Comment:
why don't we use a concrete provider type anymore?

Reply:
we can, this was already generic when I started working on this PR, should I change it back?

Reply:
I see, that's fine then